grafana-plugin/src/containers/PluginConfigPage/PluginConfigPage.tsx

```diff
@@ -1,54 +1,241 @@
 import React from 'react';
 
-import { PluginConfigPageProps, PluginMeta } from '@grafana/data';
-import { config } from '@grafana/runtime';
-import { Field, HorizontalGroup, Input } from '@grafana/ui';
+import { css } from '@emotion/css';
+import { GrafanaTheme2, PluginConfigPageProps, PluginMeta } from '@grafana/data';
+import { Alert, Field, HorizontalGroup, Input, LoadingPlaceholder, useStyles2 } from '@grafana/ui';
 import { observer } from 'mobx-react-lite';
 import { Controller, useForm } from 'react-hook-form';
 import { OnCallPluginMetaJSONData } from 'types';
 
 import { Button } from 'components/Button/Button';
-import { getOnCallApiUrl } from 'utils/consts';
+import { CollapsibleTreeView } from 'components/CollapsibleTreeView/CollapsibleTreeView';
+import { Text } from 'components/Text/Text';
+import { ActionKey } from 'models/loader/action-keys';
+import { rootStore } from 'state/rootStore';
+import {
+  DOCS_ONCALL_OSS_INSTALL,
+  DOCS_SERVICE_ACCOUNTS,
+  getOnCallApiUrl,
+  PLUGIN_CONFIG,
+  PLUGIN_ROOT,
+  REQUEST_HELP_URL,
+} from 'utils/consts';
+import { useOnMount } from 'utils/hooks';
 import { validateURL } from 'utils/string';
+import { getIsExternalServiceAccountFeatureAvailable, getIsRunningOpenSourceVersion } from 'utils/utils';
 
 type PluginConfigFormValues = {
   onCallApiUrl: string;
 };
 
 export const PluginConfigPage = observer((props: PluginConfigPageProps<PluginMeta<OnCallPluginMetaJSONData>>) => {
-  const { handleSubmit, control, formState } = useForm<PluginConfigFormValues>({
-    mode: 'onChange',
-    defaultValues: { onCallApiUrl: getOnCallApiUrl(props.plugin.meta) },
-  });
+  const {
+    pluginStore: { verifyPluginConnection },
+  } = rootStore;
 
-  const onSubmit = (values: PluginConfigFormValues) => {
-    // eslint-disable-next-line no-console
-    console.log(values);
-  };
+  useOnMount(verifyPluginConnection);
 
   return (
-    <form onSubmit={handleSubmit(onSubmit)}>
-      <Controller
-        name={'onCallApiUrl'}
-        control={control}
-        rules={{ required: 'OnCall API URL is required', validate: validateURL }}
-        render={({ field }) => (
-          <Field
-            key={'Name'}
-            label={'OnCall API URL'}
-            invalid={Boolean(formState.errors.onCallApiUrl)}
-            error={formState.errors.onCallApiUrl?.message}
-          >
-            <Input {...field} placeholder={'OnCall API URL'} data-testid="oncall-api-url-input" />
-          </Field>
-        )}
-      />
-      <HorizontalGroup>
-        <Button type="submit" disabled={!formState.isValid}>
-          Test & Save connection
-        </Button>
-        {config.featureToggles.externalServiceAccounts && <Button variant="secondary">Recreate service account</Button>}
-      </HorizontalGroup>
-    </form>
+    <>
```

Comment:
shouldn't this be wrapped in like a `VerticalGroup` at least?

Reply:
Doesn't harm, let me do that. It gives a bit more spaces between particular parts

Reply:
done